latest 1.14.0 release fails to build on armv7hl

Hi,

On Mageia Cauldron latest 1.14.0 release fails to build on armv7hl with the following build issue:

> FAILED: lib/libPocoDataODBC.so.110 
: && /usr/bin/g++ -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -march=armv7-a -mfpu=vfpv3-d16 -mtune=generic-armv7-a -mabi=aapcs-linux -mfloat-abi=hard -DNDEBUG  -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -Wl,-O1 -Wl,--build-id=sha1 -Wl,--enable-new-dtags -specs=/usr/lib/rpm/redhat/redhat-hardened-ld   -Wl,--dependency-file=Data/ODBC/CMakeFiles/DataODBC.dir/link.d -shared -Wl,-soname,libPocoDataODBC.so.110 -o lib/libPocoDataODBC.so.110 Data/ODBC/CMakeFiles/DataODBC.dir/src/Binder.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/ConnectionHandle.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Connector.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/EnvironmentHandle.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Extractor.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/ODBCException.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/ODBCMetaColumn.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/ODBCStatementImpl.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Parameter.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Preparator.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/SessionImpl.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/TypeInfo.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Unicode.cpp.o Data/ODBC/CMakeFiles/DataODBC.dir/src/Utility.cpp.o  -Wl,-rpath,/home/iurt/rpmbuild/BUILD/poco-1.14.0-build/poco-poco-1.14.0-release/build/lib:  lib/libPocoData.so.110  /usr/lib/libodbc.so  lib/libPocoFoundation.so.110  /lib/libpcre2-8.so  /usr/lib/libz.so  /usr/lib/libutf8proc.so  -lpthread  -ldl  -lrt && :
/usr/bin/ld: Data/ODBC/CMakeFiles/DataODBC.dir/src/ConnectionHandle.cpp.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for Poco::Data::ODBC::HandleException<void*, (short)1>'; Data/ODBC/CMakeFiles/DataODBC.dir/src/Binder.cpp.o:(.data.rel.ro+0x0): first defined here

Attchaed full build log:
[build.armv7hl.0.20250106193405.log](https://github.com/user-attachments/files/18335050/build.armv7hl.0.20250106193405.log)

--- a/Data/ODBC/Makefile
+++ b/Data/ODBC/Makefile
@@ -10,7 +10,8 @@ include ODBC.make
 
 objects = Binder ConnectionHandle Connector EnvironmentHandle \
 	Extractor ODBCException ODBCMetaColumn ODBCStatementImpl \
-	Parameter Preparator SessionImpl TypeInfo Unicode Utility
+	Parameter Preparator SessionImpl TypeInfo Unicode Utility \
+	Diagnostics Error 
 
 target_includes = $(POCO_BASE)/Data/testsuite/include
 

--- a/Data/ODBC/ODBC_VS90.vcproj
+++ b/Data/ODBC/ODBC_VS90.vcproj
@@ -636,6 +636,14 @@
 					>
 				</File>
 				<File
+					RelativePath=".\src\Diagnostics.cpp"
+					>
+				</File>
+				<File
+					RelativePath=".\src\Error.cpp"
+					>
+				</File>
+				<File
 					RelativePath=".\src\Unicode_UNIXODBC.cpp"
 					>
 					<FileConfiguration

--- a/Data/ODBC/ODBC_vs160.vcxproj
+++ b/Data/ODBC/ODBC_vs160.vcxproj
@@ -677,7 +677,17 @@
       <LanguageStandard>stdcpp17</LanguageStandard>
       <LanguageStandard_C>stdc11</LanguageStandard_C>
     </ClCompile>
-    <ClCompile Include="src\Unicode_UNIXODBC.cpp">
+	<ClCompile Include="src\Diagnostics.cpp">
+	  <MultiProcessorCompilation>true</MultiProcessorCompilation>
+	  <LanguageStandard>stdcpp17</LanguageStandard>
+	  <LanguageStandard_C>stdc11</LanguageStandard_C>
+	</ClCompile>
+	<ClCompile Include="src\Error.cpp">
+	  <MultiProcessorCompilation>true</MultiProcessorCompilation>
+	  <LanguageStandard>stdcpp17</LanguageStandard>
+	  <LanguageStandard_C>stdc11</LanguageStandard_C>
+	</ClCompile>
+	<ClCompile Include="src\Unicode_UNIXODBC.cpp">
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|Win32'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|x64'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_static_md|Win32'">true</ExcludedFromBuild>

--- a/Data/ODBC/ODBC_vs160.vcxproj.filters
+++ b/Data/ODBC/ODBC_vs160.vcxproj.filters
@@ -122,6 +122,12 @@
     <ClCompile Include="src\Utility.cpp">
       <Filter>ODBC\Source Files</Filter>
     </ClCompile>
+	<ClCompile Include="src\Diagnostics.cpp">
+	  <Filter>ODBC\Source Files</Filter>
+	</ClCompile>
+	<ClCompile Include="src\Error.cpp">
+	  <Filter>ODBC\Source Files</Filter>
+	</ClCompile>
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\..\DLLVersion.rc" />

--- a/Data/ODBC/ODBC_vs170.vcxproj
+++ b/Data/ODBC/ODBC_vs170.vcxproj
@@ -969,7 +969,17 @@
       <LanguageStandard>stdcpp17</LanguageStandard>
       <LanguageStandard_C>stdc11</LanguageStandard_C>
     </ClCompile>
-    <ClCompile Include="src\Unicode_UNIXODBC.cpp">
+	<ClCompile Include="src\Diagnostics.cpp">
+	  <MultiProcessorCompilation>true</MultiProcessorCompilation>
+	  <LanguageStandard>stdcpp17</LanguageStandard>
+	  <LanguageStandard_C>stdc11</LanguageStandard_C>
+	</ClCompile>
+	<ClCompile Include="src\Error.cpp">
+	  <MultiProcessorCompilation>true</MultiProcessorCompilation>
+	  <LanguageStandard>stdcpp17</LanguageStandard>
+	  <LanguageStandard_C>stdc11</LanguageStandard_C>
+	</ClCompile>
+	<ClCompile Include="src\Unicode_UNIXODBC.cpp">
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|ARM64'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|Win32'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|x64'">true</ExcludedFromBuild>

--- a/Data/ODBC/ODBC_vs170.vcxproj.filters
+++ b/Data/ODBC/ODBC_vs170.vcxproj.filters
@@ -122,6 +122,12 @@
     <ClCompile Include="src\Utility.cpp">
       <Filter>ODBC\Source Files</Filter>
     </ClCompile>
+	<ClCompile Include="src\Diagnostics.cpp">
+	  <Filter>ODBC\Source Files</Filter>
+	</ClCompile>
+	<ClCompile Include="src\Error.cpp">
+	  <Filter>ODBC\Source Files</Filter>
+	</ClCompile>
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\..\DLLVersion.rc" />

--- a/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h
+++ b/Data/ODBC/include/Poco/Data/ODBC/Diagnostics.h
@@ -19,6 +19,7 @@
 
 
 #include "Poco/Data/ODBC/ODBC.h"
+#include "Poco/Data/ODBC/Utility.h"
 #include <vector>
 #include <cstring>
 #ifdef POCO_OS_FAMILY_WINDOWS
@@ -243,10 +244,19 @@ private:
 
 // explicit instantiation definition
 #ifndef POCO_DOC
-template class Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
-template class Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
-template class Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
-template class Diagnostics<SQLHDESC, SQL_HANDLE_DESC>;
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
+extern template class Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class Diagnostics<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+extern template class ODBC_API Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
+extern template class ODBC_API Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class ODBC_API Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class ODBC_API Diagnostics<SQLHDESC, SQL_HANDLE_DESC>;
+#endif
+
 #endif
 
 

--- a/Data/ODBC/include/Poco/Data/ODBC/Error.h
+++ b/Data/ODBC/include/Poco/Data/ODBC/Error.h
@@ -126,12 +126,20 @@ private:
 
 // explicit instantiation definition
 #ifndef POCO_DOC
-template class Error<SQLHENV, SQL_HANDLE_ENV>;
-template class Error<SQLHDBC, SQL_HANDLE_DBC>;
-template class Error<SQLHSTMT, SQL_HANDLE_STMT>;
-template class Error<SQLHDESC, SQL_HANDLE_DESC>;
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class Error<SQLHENV, SQL_HANDLE_ENV>;
+extern template class Error<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class Error<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class Error<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+extern template class ODBC_API Error<SQLHENV, SQL_HANDLE_ENV>;
+extern template class ODBC_API Error<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class ODBC_API Error<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class ODBC_API Error<SQLHDESC, SQL_HANDLE_DESC>;
 #endif
 
+#endif
 
 using EnvironmentError = Error<SQLHENV, SQL_HANDLE_ENV>;
 using ConnectionError = Error<SQLHDBC, SQL_HANDLE_DBC>;

--- a/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h
+++ b/Data/ODBC/include/Poco/Data/ODBC/ODBCException.h
@@ -140,10 +140,19 @@ private:
 
 // explicit instantiation definition
 #ifndef POCO_DOC
-template class HandleException<SQLHENV, SQL_HANDLE_ENV>;
-template class HandleException<SQLHDBC, SQL_HANDLE_DBC>;
-template class HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
-template class HandleException<SQLHDESC, SQL_HANDLE_DESC>;
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class HandleException<SQLHENV, SQL_HANDLE_ENV>;
+extern template class HandleException<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class HandleException<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+extern template class ODBC_API HandleException<SQLHENV, SQL_HANDLE_ENV>;
+extern template class ODBC_API HandleException<SQLHDBC, SQL_HANDLE_DBC>;
+extern template class ODBC_API HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
+extern template class ODBC_API HandleException<SQLHDESC, SQL_HANDLE_DESC>;
+#endif
+
 #endif
 
 

--- a/Data/ODBC/src/Diagnostics.cpp
+++ b/Data/ODBC/src/Diagnostics.cpp
@@ -1,0 +1,40 @@
+//
+// Diagnostics.cpp
+//
+// Library: Data/ODBC
+// Package: ODBC
+// Module:  Diagnostics
+//
+// Copyright (c) 2025, Applied Informatics Software Engineering GmbH.
+// and Contributors.
+//
+// SPDX-License-Identifier:	BSL-1.0
+//
+
+
+#include "Poco/Data/ODBC/Diagnostics.h"
+
+
+namespace Poco {
+namespace Data {
+namespace ODBC {
+
+// explicit instantiation definition
+#ifndef POCO_DOC
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class ODBC_API Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
+template class ODBC_API Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
+template class ODBC_API Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
+template class ODBC_API Diagnostics<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+template class Diagnostics<SQLHENV, SQL_HANDLE_ENV>;
+template class Diagnostics<SQLHDBC, SQL_HANDLE_DBC>;
+template class Diagnostics<SQLHSTMT, SQL_HANDLE_STMT>;
+template class Diagnostics<SQLHDESC, SQL_HANDLE_DESC>;
+#endif
+
+#endif
+
+
+} } } // namespace Poco::Data::ODBC

--- a/Data/ODBC/src/Error.cpp
+++ b/Data/ODBC/src/Error.cpp
@@ -1,0 +1,40 @@
+//
+// Error.cpp
+//
+// Library: Data/ODBC
+// Package: ODBC
+// Module:  Error
+//
+// Copyright (c) 2025, Applied Informatics Software Engineering GmbH.
+// and Contributors.
+//
+// SPDX-License-Identifier:	BSL-1.0
+//
+
+
+#include "Poco/Data/ODBC/Error.h"
+
+
+namespace Poco {
+namespace Data {
+namespace ODBC {
+
+// explicit instantiation definition
+#ifndef POCO_DOC
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class ODBC_API Error<SQLHENV, SQL_HANDLE_ENV>;
+template class ODBC_API Error<SQLHDBC, SQL_HANDLE_DBC>;
+template class ODBC_API Error<SQLHSTMT, SQL_HANDLE_STMT>;
+template class ODBC_API Error<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+template class Error<SQLHENV, SQL_HANDLE_ENV>;
+template class Error<SQLHDBC, SQL_HANDLE_DBC>;
+template class Error<SQLHSTMT, SQL_HANDLE_STMT>;
+template class Error<SQLHDESC, SQL_HANDLE_DESC>;
+#endif
+
+#endif
+
+
+} } } // namespace Poco::Data::ODBC

--- a/Data/ODBC/src/ODBCException.cpp
+++ b/Data/ODBC/src/ODBCException.cpp
@@ -20,6 +20,17 @@ namespace Poco {
 namespace Data {
 namespace ODBC {
 
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class ODBC_API HandleException<SQLHENV, SQL_HANDLE_ENV>;
+template class ODBC_API HandleException<SQLHDBC, SQL_HANDLE_DBC>;
+template class ODBC_API HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
+template class ODBC_API HandleException<SQLHDESC, SQL_HANDLE_DESC>;
+#else
+template class HandleException<SQLHENV, SQL_HANDLE_ENV>;
+template class HandleException<SQLHDBC, SQL_HANDLE_DBC>;
+template class HandleException<SQLHSTMT, SQL_HANDLE_STMT>;
+template class HandleException<SQLHDESC, SQL_HANDLE_DESC>;
+#endif
 
 POCO_IMPLEMENT_EXCEPTION(ODBCException, Poco::Data::DataException, "Generic ODBC error")
 POCO_IMPLEMENT_EXCEPTION(InsufficientStorageException, ODBCException, "Insufficient storage error")

--- a/Foundation/Foundation_vs160.vcxproj
+++ b/Foundation/Foundation_vs160.vcxproj
@@ -977,8 +977,11 @@
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='release_static_mt|Win32'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='release_static_mt|x64'">true</ExcludedFromBuild>
     </ClCompile>
-    <ClCompile Include="src\FileStreamRWLock.cpp" />
-    <ClCompile Include="src\FileStreamRWLock_POSIX.cpp">
+	<ClCompile Include="src\BufferedBidirectionalStreamBuf.cpp" />
+	<ClCompile Include="src\BufferedStreamBuf.cpp" />
+	<ClCompile Include="src\UnbufferedStreamBuf.cpp" />
+	<ClCompile Include="src\FileStreamRWLock.cpp" />
+	<ClCompile Include="src\FileStreamRWLock_POSIX.cpp">
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|Win32'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|x64'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_static_md|Win32'">true</ExcludedFromBuild>

--- a/Foundation/Foundation_vs170.vcxproj
+++ b/Foundation/Foundation_vs170.vcxproj
@@ -1415,7 +1415,10 @@
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='release_static_mt|x64'">true</ExcludedFromBuild>
       <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='release_static_mt|ARM64'">true</ExcludedFromBuild>
     </ClCompile>
-    <ClCompile Include="src\FileStreamRWLock.cpp" />
+	<ClCompile Include="src\BufferedBidirectionalStreamBuf.cpp" />
+	<ClCompile Include="src\BufferedStreamBuf.cpp" />
+	<ClCompile Include="src\UnbufferedStreamBuf.cpp" />
+	<ClCompile Include="src\FileStreamRWLock.cpp" />
     <ClCompile Include="src\FileStreamRWLock_POSIX.cpp">
     <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|Win32'">true</ExcludedFromBuild>
     <ExcludedFromBuild Condition="'$(Configuration)|$(Platform)'=='debug_shared|x64'">true</ExcludedFromBuild>

--- a/Foundation/Makefile
+++ b/Foundation/Makefile
@@ -34,7 +34,8 @@ objects = ArchiveStrategy Ascii ASCIIEncoding AsyncChannel AsyncNotificationCent
 	FileStreamFactory URIStreamFactory URIStreamOpener UTF32Encoding UTF16Encoding UTF8Encoding UTF8String \
 	Unicode UnicodeConverter Windows1250Encoding Windows1251Encoding Windows1252Encoding \
 	UUID UUIDGenerator Void Var VarHolder VarIterator VarVisitor Format Pipe PipeImpl PipeStream SharedMemory \
-	MemoryStream FileStream AtomicCounter DataURIStream DataURIStreamFactory FileStreamRWLock
+	MemoryStream FileStream AtomicCounter DataURIStream DataURIStreamFactory FileStreamRWLock \
+	BufferedBidirectionalStreamBuf BufferedStreamBuf UnbufferedStreamBuf
 
 zlib_objects = adler32 compress crc32 deflate \
 	infback inffast inflate inftrees trees zutil

--- a/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h
+++ b/Foundation/include/Poco/BufferedBidirectionalStreamBuf.h
@@ -181,15 +181,14 @@ private:
 //
 // We provide an instantiation for char.
 //
-// Visual C++ needs a workaround - explicitly importing the template
-// instantiation - to avoid duplicate symbols due to multiple
-// instantiations in different libraries.
-//
-#if defined(_MSC_VER) && defined(POCO_DLL) && !defined(Foundation_EXPORTS)
-template class Foundation_API BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
+#else
+extern template class Foundation_API BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
 #endif
-using BufferedBidirectionalStreamBuf
-	= BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
+
+using BufferedBidirectionalStreamBuf = BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
 
 } // namespace Poco
 

--- a/Foundation/include/Poco/BufferedStreamBuf.h
+++ b/Foundation/include/Poco/BufferedStreamBuf.h
@@ -160,18 +160,17 @@ private:
 	BasicBufferedStreamBuf& operator = (const BasicBufferedStreamBuf&);
 };
 
-
 //
 // We provide an instantiation for char.
 //
-// Visual C++ needs a workaround - explicitly importing the template
-// instantiation - to avoid duplicate symbols due to multiple
-// instantiations in different libraries.
-//
-#if defined(_MSC_VER) && defined(POCO_DLL) && !defined(Foundation_EXPORTS)
-template class Foundation_API BasicBufferedStreamBuf<char, std::char_traits<char>>;
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class BasicBufferedStreamBuf<char, std::char_traits<char>>;
+#else
+extern template class Foundation_API BasicBufferedStreamBuf<char, std::char_traits<char>>;
 #endif
-typedef BasicBufferedStreamBuf<char, std::char_traits<char>> BufferedStreamBuf;
+
+using BufferedStreamBuf = BasicBufferedStreamBuf<char, std::char_traits<char>>;
 
 
 } // namespace Poco

--- a/Foundation/include/Poco/UnbufferedStreamBuf.h
+++ b/Foundation/include/Poco/UnbufferedStreamBuf.h
@@ -163,15 +163,14 @@ private:
 //
 // We provide an instantiation for char.
 //
-// Visual C++ needs a workaround - explicitly importing the template
-// instantiation - to avoid duplicate symbols due to multiple
-// instantiations in different libraries.
-//
-#if defined(_MSC_VER) && defined(POCO_DLL) && !defined(Foundation_EXPORTS)
-template class Foundation_API BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
-#endif
-typedef BasicUnbufferedStreamBuf<char, std::char_traits<char>> UnbufferedStreamBuf;
 
+#if defined(POCO_OS_FAMILY_WINDOWS)
+extern template class BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
+#else
+extern template class Foundation_API BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
+#endif
+
+using UnbufferedStreamBuf = BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
 
 } // namespace Poco
 

--- a/Foundation/src/BufferedBidirectionalStreamBuf.cpp
+++ b/Foundation/src/BufferedBidirectionalStreamBuf.cpp
@@ -1,0 +1,25 @@
+//
+// BufferedBidirectionalStreamBuf.cpp
+//
+// Library: Foundation
+// Package: Streams
+// Module:  StreamBuf
+//
+// Copyright (c) 2025, Applied Informatics Software Engineering GmbH.
+// and Contributors.
+//
+// SPDX-License-Identifier:	BSL-1.0
+//
+
+
+#include "Poco/BufferedBidirectionalStreamBuf.h"
+
+namespace Poco {
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class Foundation_API BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
+#else
+template class BasicBufferedBidirectionalStreamBuf<char, std::char_traits<char>>;
+#endif
+
+}

--- a/Foundation/src/BufferedStreamBuf.cpp
+++ b/Foundation/src/BufferedStreamBuf.cpp
@@ -1,0 +1,25 @@
+//
+// BufferedStreamBuf.cpp
+//
+// Library: Foundation
+// Package: Streams
+// Module:  StreamBuf
+//
+// Copyright (c) 2025, Applied Informatics Software Engineering GmbH.
+// and Contributors.
+//
+// SPDX-License-Identifier:	BSL-1.0
+//
+
+
+#include "Poco/BufferedStreamBuf.h"
+
+namespace Poco {
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class Foundation_API BasicBufferedStreamBuf<char, std::char_traits<char>>;
+#else
+template class BasicBufferedStreamBuf<char, std::char_traits<char>>;
+#endif
+
+}

--- a/Foundation/src/FileStream.cpp
+++ b/Foundation/src/FileStream.cpp
@@ -13,7 +13,6 @@
 
 
 #include "Poco/FileStream.h"
-#include "Poco/Exception.h"
 #if defined(POCO_OS_FAMILY_WINDOWS)
 #include "FileStream_WIN32.cpp"
 #else

--- a/Foundation/src/UnbufferedStreamBuf.cpp
+++ b/Foundation/src/UnbufferedStreamBuf.cpp
@@ -1,0 +1,25 @@
+//
+// UnbufferedStreamBuf.cpp
+//
+// Library: Foundation
+// Package: Streams
+// Module:  StreamBuf
+//
+// Copyright (c) 2025, Applied Informatics Software Engineering GmbH.
+// and Contributors.
+//
+// SPDX-License-Identifier:	BSL-1.0
+//
+
+
+#include "Poco/UnbufferedStreamBuf.h"
+
+namespace Poco {
+
+#if defined(POCO_OS_FAMILY_WINDOWS)
+template class Foundation_API BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
+#else
+template class BasicUnbufferedStreamBuf<char, std::char_traits<char>>;
+#endif
+
+}

